src/middlewared/middlewared/etc_files/syslogd.py

```diff
@@ -9,6 +9,53 @@
 logger = logging.getLogger(__name__)
 
 
+def generate_syslog_remote_destination(middleware, advanced_config):
+    result = ""
+    if advanced_config["syslogserver"]:
+        if ":" in advanced_config["syslogserver"]:
+            host, port = advanced_config["syslogserver"].split(":")[:2]
+        else:
+            host, port = advanced_config["syslogserver"], "514"
+
+        result += 'destination loghost { '
+
+        if advanced_config["syslog_transport"] == "TLS":
+            try:
+                certificate = middleware.call_sync(
+                    "certificate.query",
+                    [
+                        ("id", "=", advanced_config["syslog_tls_certificate"]),
+                        ("revoked", "=", False),
```

Comment:
An alert should be generated when a certificate is revoked which is in use by a service. We should make amends in `certificates.py` for alert source to reflect this.